module.nix

```diff
@@ -1,37 +1,119 @@
 # configuration
-{ images
-, linuxSystem
+{
+  image,
+  linuxSystem,
+}: {
+  config,
+  lib,
+  pkgs,
+  ...
 }:
-{ config, lib, pkgs, ... }: {
+with lib; {
   options.nix-rosetta-builder = {
-    enable = (lib.mkEnableOption "Nix Rosetta Linux builder") // { default = true; };
+    enable =
+      (mkEnableOption "Nix Rosetta Linux builder")
+      // {
+        default = true;
+      };
+
+    config = mkOption {
+      type = types.deferredModule;
+      default = {};
+      description = ''
+        Extra NixOS configuration options for the VM. This is merged with
+        the default configuration. Default values will be overridden if
+        specified here. Changes will cause a rebuild of the VM image.
+      '';
+      example = literalExpression ''
+        ({ pkgs, ... }:
+        {
+          environment.systemPackages = [ pkgs.neovim ];
+        })
+      '';
+    };
+
+    cores = mkOption {
+      type = types.int;
+      default = 8;
+      description = ''
+        The number of CPU cores allocated to the Lima instance.
+        This also sets the maximum number of jobs allowed for the
+        builder in the `nix.buildMachines` specification.
+      '';
+    };
+
+    debug = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enable root access in VM and debug logging.
+      '';
+    };
+
+    diskSize = mkOption {
+      type = types.str;
+      default = "100GiB";
+      description = ''
+        The size of the disk image for the Lima instance.
+      '';
+    };
+
+    enableRosetta = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Enable Rosetta 2 in the VM, allowing cross
+        compilation of `x86_64-linux` packages.
+      '';
+    };
 
-    onDemand = lib.mkOption {
-      type = lib.types.bool;
+    memory = mkOption {
+      type = types.str;
+      default = "6GiB";
+      description = ''
+        The amount of memory to allocate to the VM.
+      '';
+      example = "8GiB";
+    };
+
+    onDemand = mkOption {
+      type = types.bool;
       default = false;
       description = ''
         FIXME
       '';
     };
+
+    port = mkOption {
+      type = types.int;
+      default = 31122;
+      description = ''
+        The SSH port used by the VM.
+      '';
+    };
   };
 
-  config =
-  let
-    inherit (import ./constants.nix)
+  config = let
+    inherit
+      (import ./constants.nix)
       name
       linuxHostName
       linuxUser
-
       sshKeyType
       sshHostPrivateKeyFileName
       sshHostPublicKeyFileName
       sshUserPrivateKeyFileName
       sshUserPublicKeyFileName
-
-      debug
       ;
+
+    imageWithFinalConfig = image.override {
```

Comment:
This is so much better than the previous two-target, `default`/`on-demand` method.